src/OpenTelemetry.Extensions.Hosting/Implementation/SelfDiagnosticsLoggingHostedService.cs

```diff
@@ -0,0 +1,54 @@
+// <copyright file="SelfDiagnosticsLoggingHostedService.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using OpenTelemetry.Internal;
+
+namespace OpenTelemetry.Extensions.Hosting.Implementation
+{
+    internal class SelfDiagnosticsLoggingHostedService : IHostedService
```

Comment:
Adding background hosted services is not supported in some environments like Azure functions. (https://github.com/open-telemetry/opentelemetry-dotnet/pull/1672#discussion_r551650928)

The original reason why we used IHostedService was to have someone try to retrieve "TracerProvider" from DI, to trigger its creation. It *might* be possible for us to get rid of IHostedService usage in that scenario.

Adding more will make it difficult.

Reply:
I'd prefer to have this just be added as an instrumentation in the TracerProvider and have the same lifetime as that class. But unfortunately I can't access the IServiceProvider in order to get access to the ILoggerFactory from the TracerProviderBuilder. Which is why I was asking if you would be interested in a PR for that. https://github.com/open-telemetry/opentelemetry-dotnet/issues/894